### PR TITLE
Opgops 1706

### PR DESCRIPTION
**Pull request by tim-mcarthur-moj**
allow running of new shared ansible playbooks

### CODE REVIEW

File: cotton/ansible/ansibleutils.py [hunk 1]
```diff
@@ -5,15 +5,24 @@
...
         :param roles_version:
         :param destroy_stack:
+        :param playbooks_version:
+        :param playbook_name:
+        :param playbook_path:
         :return:
         """
         # add extra vars to this string
-        extra_vars = 'target={} opg_ansible_version={}'.format(env.stackname, roles_version)
+        extra_vars = 'target={} opg_ansible_version={} opg_playbooks_version={}'.format(env.stackname, roles_version, playbooks_version)
```
Comment (brettminnie) [inline cotton/ansible/ansibleutils.py]: split this line please on the format, pep8 will have a frothy

File: cotton/ansible/ansibleutils.py [hunk 2]
```diff
@@ -22,4 +31,11 @@ def run_ansible_playbook(roles_version='master', destroy_stack=False):
             # checkout roles from opg-ansible-roles repo
             local('ansible-playbook -i hosts site.yml -e "' + extra_vars + '"')
             # run provisioning playbook
-            local('ansible-playbook -i hosts provision.yml -e "' + extra_vars + '" -v')
+            playbook_cmd = 'ansible-playbook -i hosts '
+            cmd_suffix = ' -e "{}" -v'.format(extra_vars)
+            if playbook_name is None:
```
Comment (brettminnie) [inline cotton/ansible/ansibleutils.py]: should we not default to 
```python
provision_cmd = 'provision.yml '

if playbook_name is not None:

```

File: cotton/ansible/ansibleutils.py [hunk 3]
```diff
@@ -22,4 +31,11 @@ def run_ansible_playbook(roles_version='master', destroy_stack=False):
             # checkout roles from opg-ansible-roles repo
             local('ansible-playbook -i hosts site.yml -e "' + extra_vars + '"')
             # run provisioning playbook
-            local('ansible-playbook -i hosts provision.yml -e "' + extra_vars + '" -v')
+            playbook_cmd = 'ansible-playbook -i hosts '
+            cmd_suffix = ' -e "{}" -v'.format(extra_vars)
```
Comment (carinadigital) [inline cotton/ansible/ansibleutils.py]: What happens if extra_vars is empty?

Comment (brettminnie) [inline cotton/ansible/ansibleutils.py]: Good spot, need a guard to check it has a value

**Pull request merged**